apps/modernization-ui/src/pages/patient/profile/addresses/AddressesTable.tsx

```diff
@@ -278,59 +276,73 @@ export const AddressesTable = ({ patient }: Props) => {
-                            Add address
-                        </Button>
-                    </div>
-                }
-                tableHeader={'Address'}
-                tableHead={tableHead}
-                tableBody={addresses?.map((name, index: number) => (
-                    <tr key={index}>
-                        <td className={`font-sans-md table-data ${tableHead[0].sort !== 'all' && 'sort-td'}`}>
-                            {name?.asOf ? (
-                                <span>
-                                    {format(new Date(name?.asOf), 'MM/dd/yyyy')} <br />{' '}
-                                </span>
-                            ) : (
-                                <NoData />
-                            )}
-                        </td>
-                        <td className={`font-sans-md table-data ${tableHead[1].sort !== 'all' && 'sort-td'}`}>
-                            {name?.type ? (
-                                <span>
-                                    {name?.type.description}
-                                    {name.use?.description ? `/${name.use?.description}` : ''}
-                                </span>
-                            ) : (
-                                <NoData />
-                            )}
-                        </td>
-                        <td className={`font-sans-md table-data ${tableHead[2].sort !== 'all' && 'sort-td'}`}>
-                            {name?.address1 || name?.address2 ? <span>{getAddress(name)}</span> : <NoData />}
-                        </td>
-                        <td className={`font-sans-md table-data ${tableHead[3].sort !== 'all' && 'sort-td'}`}>
-                            {name?.city ? <span>{name?.city}</span> : <NoData />}
-                        </td>
-                        <td className={`font-sans-md table-data ${tableHead[4].sort !== 'all' && 'sort-td'}`}>
-                            {name?.state ? <span>{name?.state?.description}</span> : <NoData />}
-                        </td>
-                        <td className={`font-sans-md table-data ${tableHead[5].sort !== 'all' && 'sort-td'}`}>
-                            {name?.zipcode ? <span>{name?.zipcode}</span> : <NoData />}
-                        </td>
-                        <td>
+    /**
+     * Formats the Address object into TableComponent compatible TableBody object which represents a single row.
+     * Each "title" in the tableDetails is a template of each column cell of the row being created.
+     * @param {PatientAddress} address, each item of the morbidity response data
+     * @param {number} index, index of the array item
+     * @return {TableBody}
+     */
+    const generateTableRow = (address: PatientAddress, index: number): TableBody => {
+        return {
+            id: index,
+            tableDetails: [
+                {
+                    id: 1,
+                    title: address?.asOf ? (
```

Comment:
question: Are the extra `<span>` elements required?  The `font-sans-1xs` class is setting the font-size should be standard for all tables and the `table-data` class is not used by the `TableComponent`.

Reply:
Oh I haven't changed the template much. Fix all those now. Thanks